src/postamble.js

```diff
@@ -44,8 +43,15 @@ if (memoryInitializer) {
   } else if (Module['memoryInitializerPrefixURL']) {
     memoryInitializer = Module['memoryInitializerPrefixURL'] + memoryInitializer;
   }
-  if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
-    var data = Module['readBinary'](memoryInitializer);
+  var data = Module['memoryInitializerBytes'];
+  if (!data) {
+    if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
+      data = Module['readBinary'](memoryInitializer);
+    } else {
+      data = tryParseAsDataURI(memoryInitializer);
```

Comment:
Yep. `tryParseAsDataURI` returns `undefined` if the input isn't a valid data URI, causing the normal browser XHR loading steps in the `else` statement to run.